api/api/controllers/agents_controller.py

```diff
@@ -424,57 +442,79 @@ async def put_upgrade_agent(request, agent_id, pretty=False, wait_for_complete=F
                           f_kwargs=remove_nones_to_dict(f_kwargs),
                           request_type='distributed_master',
                           is_async=False,
-                          wait_for_complete=True,  # Force wait_for_complete until timeout problems are resolved
+                          wait_for_complete=wait_for_complete,
                           logger=logger,
+                          broadcasting=agents_list == '*',
                           rbac_permissions=request['token_info']['rbac_policies']
                           )
     data = raise_if_exc(await dapi.distribute_function())
 
     return web.json_response(data=data, status=200, dumps=prettify if pretty else dumps)
 
 
-async def put_upgrade_custom_agent(request, agent_id, pretty=False, wait_for_complete=False, file_path=None,
-                                   installer=None):
-    """Upgrade agent using a local WPK file.'.
+async def put_upgrade_custom_agents(request, agents_list='*', pretty=False, wait_for_complete=False,
+                                    file_path=None, installer=None):
+    """Upgrade agents using a local WPK file.
 
-    :param pretty: Show results in human-readable format
-    :param wait_for_complete: Disable timeout response
-    :param agent_id: Agent ID. All posible values since 000 onwards.
-    :param file_path: Path to the WPK file. The file must be on a folder on the Wazuh's installation directory
-    (by default, <code>/var/ossec</code>).
-    :type installer: str
-    :return: ApiResponse
+    Parameters
+    ----------
+    pretty : bool
+        Show results in human-readable format.
+    wait_for_complete : bool
+        Disable timeout response.
+    agents_list : list
+        List of agent IDs. All possible values from 000 onwards.
+    file_path : str
+        Path to the WPK file. The file must be on a folder on the Wazuh's installation directory (by default, <code>/var/ossec</code>).
+    installer : str
+        Installation file.
+
+    Returns
+    -------
+    ApiResponse
+        Upgrade message after trying to upgrade the agents.
     """
-    f_kwargs = {'agent_list': [agent_id],
+    if len(agents_list) > 100:
```

Comment:
Why use this instead of maxItems limitation for array in spec.yaml?

Reply:
I think we talked about using that if clause there with a new exception. Changed to maxItems anyway as it is a better way to solve it